src/lwc/geGiftEntryFormApp/geGiftEntryFormApp.js

```diff
@@ -190,34 +209,74 @@ export default class GeGiftEntryFormApp extends NavigationMixin(LightningElement
             const purchaseResponse = await this.makePurchaseCall();
             if (purchaseResponse) {
 
-                this.dataImportRecord[PAYMENT_STATUS__C] = this.getPaymentStatus(purchaseResponse);
-                this.dataImportRecord[PAYMENT_DECLINED_REASON__C] =
-                    this.getPaymentDeclinedReason(purchaseResponse);
+                let errors = this.processPurchaseResponse(purchaseResponse);
 
                 this.dataImportRecord = await upsertDataImport({ dataImport: this.dataImportRecord });
 
-                this.isFailedPurchase = purchaseResponse.statusCode !== 201;
-                if (this.isFailedPurchase) {
-                    let errors = this.getFailedPurchaseMessage(purchaseResponse);
+                if (isNotEmpty(errors)) {
+                    this.isFailedPurchase = true;
+
                     let labelReplacements = [this.CUSTOM_LABELS.commonPaymentServices, errors];
                     let formattedErrorResponse = format(this.CUSTOM_LABELS.gePaymentProcessError, labelReplacements);
 
                     // We use the hex value for line feed (new line) 0x0A
                     let splitErrorResponse = formattedErrorResponse.split(LABEL_NEW_LINE);
-                    
+
                     const form = this.template.querySelector('c-ge-form-renderer');
                     form.showSpinner = false;
                     fireEvent(null, 'paymentError', {
                         error: {
                             message: splitErrorResponse,
                             isObject: true
                         }
-                     });
+                    });
```

Comment:
What do you think about pulling lines 219 - 232 into a separate method called `dispatchPaymentErrorEvent` or something along those lines?